File: platforms/kotlin/examples/core/texture_mip_chain/TextureMipChain.kt
import org.fragmentcolor.*

val renderer = Renderer()
// Encoded image bytes the caller has on hand (could come off a worker).
val png = byteArrayOf(0x89.toByte(), 0x50.toByte(), 0x4E.toByte(), 0x47.toByte(), 0x0D.toByte(), 0x0A.toByte(), 0x1A.toByte(), 0x0A.toByte())
val chain = TextureMipChain.prepare(png, TextureFormat.RGBA8_UNORM_SRGB, null)

// Upload the chain through the regular create_texture entry point.
val texture = renderer.createTexture(TextureInputMobile.Prepared(chain), null)

File: platforms/web/examples/core/texture_mip_chain/TextureMipChain.js
import { Renderer, TextureFormat, TextureMipChain } from "fragmentcolor";

const renderer = new Renderer();
// Minimal 1x1 encoded PNG bytes.
const b64 = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAADElEQVR4nGM4ceIEAAS0AlkWLoFAAAAAAElFTkSuQmCC";
const pngBytes = Uint8Array.from(atob(b64), c => c.charCodeAt(0));
// Encoded path: size=null, bytes decoded as an image.
const chain = TextureMipChain.prepare(pngBytes, TextureFormat.Rgba8UnormSrgb, null);

// Hand the chain to the unified create_texture entry - same vocabulary as
// every other texture path; From<TextureMipChain> selects the GPU-only
// upload internally.
const texture = await renderer.createTexture(chain);